checkoutmanager/runner.py

```diff
@@ -82,6 +82,25 @@ def execute_action(dirinfo, custom_actions, action):
         return e
 
 
+def run_one(action, directory=None, url=None, conf=None):
+    custom_actions = get_custom_actions()
+    if not conf:
+        conf = config.Config(os.path.expanduser(CONFIGFILE_NAME))
+    from .dirinfo import DirInfo
+    if directory:
```

Comment:
This nested if with elifs and without an else is too unclear for me. I just cannot wrap my head around it completely.

The confusion is partially because directory an also be a DirInfo and you use the `directory` name for both.

What about:

    dir_info = None
    if url:
        dir_info = conf.directory_from_path(directory
    if directory:
        if isinstance(directory, DirInfo):
             dir_info = directory
        else:
             dir_info = conf.directory_from_path(directory)
    if not dir_info:
        # perhaps print an error, but it is probably not needed.
    executor = ...